src/swar_synthesizer.py

```diff
@@ -4,82 +4,98 @@
 import struct
 import pyaudio
 
-def swar2freq(swar,basefreq):
+SWAR_FREQUENCY_MAPPER = {
+    's':{
+        'shuddh':1,
+        'vakra':1, # Error handling mechanism.
+    },
+    'r':{
+        'shuddh': 1.125,
+        'vakra': 1.058
+    },
+    'g':{
+        'shuddh': 1.266,
+        'vakra': 1.188
+    },
+    'm':{
+        'shuddh': 1.333,
+        'vakra': 1.412
+    },
+    'p':{
+        'shuddh': 1.5,
+        'vakra': 1.5 # Error handling mechanism
+    },
+    'd':{
+        'shuddh': 1.688,
+        'vakra': 1.586
+    },
+    'n':{
+        'shuddh': 1.898,
+        'vakra': 1.78
+    },
+    '0': 0,
+}
+VALID_SWAR_INPUTS_SUBSET_1 = set(['<', '>'])
+VALID_SWAR_INPUTS_SUBSET_2 = set(['s', 'r', 'g', 'm', 'p', 'd', 'n'])
```

Comment:
This definitely seems very systematic, but can you explain why this needs to be done this way? The type conversions from list to set to list back to set?

Reply:
I read somewhere that looking up entities in a `set` is faster than looking up entities in a `list`. Unfortunately, I am unable to find the blog post / stackoverflow discussion link related to this. On the other hand, the "lookup is faster in set" theory might be notable if the data size is large. In that case, this is an unnecessary premature optimization. I will revert this `list` to `set` conversion as it has introduced confusion and possibly premature optimization instead of solving a (non-existent) performance issue. Sorry for the confusion. 😅